feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -1,23 +1,83 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import {
+  AssociatedObject,
+  Category,
+  Ticket,
+} from '@spartacus/customer-ticketing/root';
 import { FormUtils } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
 import { CustomerTicketingDialogComponent } from '../../shared/customer-ticketing-dialog/customer-ticketing-dialog.component';
-
 @Component({
   selector: 'cx-customer-ticketing-create-dialog',
   templateUrl: './customer-ticketing-create-dialog.component.html',
 })
 export class CustomerTicketingCreateDialogComponent
   extends CustomerTicketingDialogComponent
-  implements OnInit
+  implements OnInit, OnDestroy
 {
+  ticketCategories$: Observable<Category[]> =
+    this.customerTicketingFacade.getTicketCategories();
+  ticketAssociatedObjects$: Observable<AssociatedObject[]> =
+    this.customerTicketingFacade.getTicketAssociatedObjects();
+  subscription: Subscription;
+
+  @Input()
+  selectedCategory: Category;
+
+  @Input()
+  selectedAssociatedTo: AssociatedObject;
+
+  protected getCreateTicketPayload(): Ticket {
+    return {
+      message: this.form?.get('message')?.value,
+      subject: this.form?.get('subject')?.value,
+      associatedTo: this.selectedAssociatedTo,
+      ticketCategory: this.selectedCategory,
+    };
+  }
+
   ngOnInit(): void {
     this.buildForm();
   }
 
+  getSelectedAssociatedObject(code: string, modifiedAt: string, type: string) {
```

Comment:
```suggestion
  setSelectedAssociatedObject(code: string, modifiedAt: string, type: string) {
```

Nothing is being returned, it is confusing to call it GET